### ISSUE TITLE
CA1305 Specify IFormatProvider misses nullable value types

### ISSUE THREAD

**Issue by jnm2** [dotnet/roslyn-analyzers#6586]
### Analyzer

**Diagnostic ID**: [CA1305](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1305): Specify IFormatProvider

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**:  7.0.300-preview.23179.2

### Describe the bug

Implicit use of CurrentCulture is just as much of an issue on `int?` as on `int`, but the analyzer doesn't show this.

Since `Nullable<T>.ToString` does not have an overload that accepts IFormatProvider, the fix should be to use null-conditional access: `y?.ToString(CultureInfo.CurrentCulture)`

### Steps To Reproduce

```cs
class C
{
    string M(int x, int? y)
    {
        // ⚠️ CA1305 The behavior of 'int.ToString()' could vary based on the current user's locale settings. Replace
        // this call in 'C.M(int, int?)' with a call to 'int.ToString(IFormatProvider)'.
        x.ToString();

        // No warning, but one should be shown.
        y.ToString();
    }
}
```

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net7.0</TargetFramework>
    <AnalysisMode>recommended</AnalysisMode>
  </PropertyGroup>

</Project>
```

**Comment by Morilli** [comment]
How would I go about disabling this analyzer for nullable types now? Previously, adding `T:System.Int32` to `excluded_symbol_names` excluded the analyzer from analyzing and reporting on `int`, but what about `int?`?

It's kind of frustrating that there seems to be no example or documentation at all about how this would be done. I tried ``T:System.Nullable``, ``T:System.Nullable`1``, ``T:System.Nullable`1[System.Int32]``, but none of these seem to work.

Any advice?

**Comment by sharwell** [comment]
@Morilli excluding `T:System.Int32` should apply to both `int` and `int?`. If this is not the case, can you file a new issue?
